modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -0,0 +1,886 @@
+        Map<String, String> responseHeaders = new HashMap<>();
+        responseHeaders.put(SCIMConstants.CONTENT_TYPE_HEADER, SCIMConstants.APPLICATION_JSON);
+        return new SCIMResponse(exception.getStatus(), encoder.encodeSCIMException(exception), responseHeaders);
+    }
+
+    private Group getNewGroup() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+
+        return decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+    }
+
+    @DataProvider(name = "dataForGetGroupSuccess")
+    public Object[][] dataToGetGroupSuccess() throws CharonException, InternalErrorException, BadRequestException {
+
+        Group group = getNewGroup();
+        String id = group.getId();
+
+        return new Object[][]{
+                {id, null, null, group},
+                {id, null, "members", group},
+                {id, "displayName", null, group},
+                {id, "displayName", "members", group}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetGroupSuccess")
+    public void testGetGroupSuccess(String id, String attributes, String excludeAttributes, Object objectGroup)
+            throws BadRequestException, NotFoundException, CharonException, NotImplementedException {
+
+        Group group = (Group) objectGroup;
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema) CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(userManager.getGroup(id, requiredAttributes)).thenReturn(group);
+        SCIMResponse scimResponse = groupResourceManager.get(id, userManager, attributes, excludeAttributes);
+        JSONObject obj = new JSONObject(scimResponse.getResponseMessage());
+        if (attributes != null && excludeAttributes != null) {
+            Assert.assertFalse(obj.has("meta"));
+            Assert.assertFalse(obj.has("members"));
+        } else if (attributes != null && excludeAttributes == null) {
+            Assert.assertFalse(obj.has("meta"));
+        } else if (attributes == null && excludeAttributes != null) {
+            Assert.assertTrue(obj.has("meta"));
+            Assert.assertFalse(obj.has("members"));
+        } else {
+            Assert.assertTrue(obj.has("meta"));
+            Assert.assertTrue(obj.has("members"));
+        }
```

Comment:
```suggestion
        if (attributes != null) {
            Assert.assertTrue(obj.has(attributes));
        }
        if (excludeAttributes != null) {
            Assert.assertFalse(obj.has(excludeAttributes));
        }
```

Does displayName has to be always there in the json?
Does availability of meta depend on the availability of `attributes` or `excludeAttributes`?

Reply:
Fixed in commit 6ee99fa6106574787ffb757a289890d2466fe80a